### DISCUSSION TITLE
New functions ideas

### DISCUSSION THREAD

**Discussion by Marow** [astubenbord/paperless-mobile#89] [Feature Requests]
Hi,

i have two ideas!

1. Auto Scan
Why dont auto "shot" the paper when the app gets in focus?

2. Fixed Crop
Most times i have the same place with same distance and everything the same for my papers to make them paperless. But sometimes the auto crop doesnt work perfect. So it would be great, if there is an option where i can setup the crop to fixed points.

For myself i have a box where i scan the papers and the phone is on a place, without moving or something.
If these two features are in the app, it makes it so much easier!

**Comment by astubenbord** [comment]
Hi, thanks for your suggestions. This is definitely something I am planning, but building a custom document scanner (e. g. using dart:ffi) is a lot of work and by far not trivial. I do not have control over the current library, which uses native modules for the edge detection. This will probably take lots of time and I'm not sure I can work on this in the near future.

**Reply by Marow** [reply]
Oh okay... I tested few document scanner before i started with paperless and the most a working a little bit better than this ones. But its okay for the overall usage. Would be great to see something like that in the future

**Reply by astubenbord** [reply]
Yes, you also have the option to use programs like office lens and then share the scanned pdf file with the app, which will result in the same as scanning from within the app, except that you will most likely get a much better scan quality. But be sure that this is one of the next bigger topics I'm looking into, because you're right, there is a lot of potential to make the scanning workflow faster and more efficient.

**Comment by astubenbord** [comment]
I know this thread is quite old, but still relevant I guess: Feel free to check out the latest pre-release with a new scanner: https://github.com/astubenbord/paperless-mobile/discussions/645 and let me know your thoughts in the discussion. The scanner is fully custom and therefore suggestions and ideas for an improved workflow can be implemented more easily.
